extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+/**
+ * Discussion user log page
+ */
+class SpecialDiscussionsLogController extends WikiaSpecialPageController {
+	// @todo
+	//1. Pagination
+	//2. Add date range input field
+
+	const KIBANA_BASEURL = 'http://query.es.service.sjc.consul:9200';
+	const PAGINATION_SIZE = 50;
+	const DAYS_RANGE = 15;
+	const HTTP_STATUS_OK = 200;
+	const NO_USER_MATCH_ERROR = "Provided username did not match any user";
+
+	static $inputFormTemplate = 'extensions/wikia/SpecialDiscussionsLog/templates/SpecialDiscussionsLog_inputForm.mustache';
+	static $userLogTemplate = 'extensions/wikia/SpecialDiscussionsLog/templates/SpecialDiscussionsLog_userLog.mustache';
+
+	public function __construct() {
+		parent::__construct( 'DiscussionsLog' );
+	}
+
+	public function execute( $subpage ) {
+
+		if ( !$this->checkAccess() ) {
+			return;
+		}
+
+		$this->setHeaders();
+
+		$this->wg->Out->clearHTML();
+		$this->wg->Out->setPageTitle( wfMessage( 'discussionslog-pagetitle' )->escaped() );
+
+		$output = $this->getInputForm();
+
+		$userName = null;
+		if ( $this->wg->request->wasPosted() ) {
+			$userName = $this->wg->request->getVal( 'username', null );
+			$output .= $this->getUserLog( $userName );
+		}
+
+		$this->wg->Out->addHtml( $output );
+	}
+
+	private function getInputForm() {
+		return \MustacheService::getInstance()->render(
+			self::$inputFormTemplate,
+			[
+				'userNameLabel' => wfMessage( 'discussionslog-username-label' )
+						->escaped(),
+				'viewLogsAction' => wfMessage( 'discussionslog-view-logs' )
+						->escaped()
+			]
+		);
+	}
+
+	private function constructKibanaUrl( $dayOffset ) {
+		$date = time() - ( $dayOffset * 24 * 60 * 60 );
+		return self::KIBANA_BASEURL . '/logstash-' . date( 'Y.m.d', $date ) . '/_search';
+	}
+
+	private function getUserIdByUsername( $userName ) {
+		if ( empty( $userName ) ) {
+			throw new InvalidArgumentException( self::NO_USER_MATCH_ERROR );
+		}
+
+		$user = User::newFromName( $userName );
+		if ( !$user || empty( $user->getId() ) ) {
+			throw new InvalidArgumentException( self::NO_USER_MATCH_ERROR );
+		}
+
+		return $user->getId();
+	}
+
+	private function getUserLog( $userName ) {
+		$displayedUserLogRecords = [];
+		$hasNoUserLogRecords = false;
+		$hasUserError = false;
+		$userId = null;
+		$userErrorMessage = null;
+
+		try {
+			$userId = $this->getUserIdByUsername( $userName );
+		} catch ( Exception $e ) {
+			$hasUserError = true;
+		}
+
+		$userLogRecords = $this->aggregateLogSearches( $userId, $userName );
+
+		if ( count( $userLogRecords ) == 0 ) {
+			$hasNoUserLogRecords = true;
```

Comment:
Agreed. Unfortunately, there seems to be a problem with this type of check in our php implementation of mustache. I created https://wikia-inc.atlassian.net/browse/MAIN-6445 to deal with it after we upgrade to the new version of the mustache extension.